packages/datatrak-web/src/layout/UserMenu/MenuList.tsx

```diff
@@ -46,69 +56,57 @@ export const MenuList = ({
   children?: ReactNode;
   onCloseMenu: () => void;
 }) => {
-  const { isLoggedIn } = useCurrentUser();
+  const { isLoggedIn, projectId } = useCurrentUser();
   const { mutate: logout } = useLogout();
 
-  // The help centre link is the same for both logged in and logged out users
-  const helpCentre = {
+  const accountSettingsItem = {
+    label: 'Account settings',
+    to: ROUTES.ACCOUNT_SETTINGS,
+  };
+  // The help centre link is the same for both logged-in and logged-out users
+  const helpCentreItem = {
     label: 'Help centre',
     href: 'https://beyond-essential.slab.com/posts/tupaia-instruction-manuals-05nke1dm',
     isExternal: true,
     component: Link,
   };
-  const onClickLogout = () => {
-    logout();
-    onCloseMenu();
+  const logOutItem = {
+    label: 'Log out',
+    onClick: () => {
+      logout();
+      onCloseMenu();
+    },
+    component: 'button',
   };
 
-  const menuItems = isLoggedIn
-    ? [
-        {
-          label: 'Account settings',
-          to: ROUTES.ACCOUNT_SETTINGS,
-        },
-        helpCentre,
-        {
-          label: 'Log out',
-          onClick: onClickLogout,
-          component: 'button',
-        },
-      ]
-    : [helpCentre];
+  const hasProjectSelected = !!projectId;
+
+  const menuItems: MenuItem[] = [];
+  if (isLoggedIn && hasProjectSelected) {
+    menuItems.push(accountSettingsItem);
+  }
```

Comment:
I would potentially say that now there are more than 2 different cases for what menu items should be there we could make a function that handles it, to nicely contain it and make it super readable for anyone else coming through the code.
So a suggestion would be something like:

```
const getMenuItems = () => {
    if(isLoggedIn) {
       return !!projectId ? [accountSettingsItem, helpCentreItem, logoutItem] : [helpCentreItem, logoutItem]
    }
return [helpCentreItem]
}

const menuItems = getMenuItems();
```

Reply:
I had thought about the nested `if`s/ternaries too, and while it’s technically more performant, I thought it was a little clearer to just append them to the array in order, each with a predicate for just that menu item. It means redundantly checking some things twice (like `isLoggedIn`), but this feels more extendible

Wrapping it in a function is a good idea, though—I’ll make that change 🙏